lib/compile/formats.js

```diff
@@ -126,7 +126,11 @@ function date_time(str) {
 function hostname(str) {
   // https://tools.ietf.org/html/rfc1034#section-3.5
   // https://tools.ietf.org/html/rfc1123#section-2
-  return str.length <= 255 && HOSTNAME.test(str);
+
+  // https://devblogs.microsoft.com/oldnewthing/?p=7873
+  // Count octets, not string length (remove any trailing dot)
+  // With starting and ending length octets this would be 255, not 253
+  return str.replace(/\.$/, '').length <= 253 && HOSTNAME.test(str);
```

Comment:
I think the comment may be the other way around? "... this would be 253, not 255"? Or am I confused?

Also using replace to remove the trailing "." with RegExp seems inefficient, it could be just:

```
str.length <= (str[str.length-1] == '.'  ? 254 : 253)
```

Reply:
The comment is correct but maybe badly worded. Your suggestion would work.